### PR TITLE
2004 show locked in UI

### LINKED ISSUE #2004: Show Locked state in UI

### PR DESCRIPTION
**Pull request by AdrianOlosutean**
Closes #2004

### CODE REVIEW

File: menas/ui/components/dataset/datasetDetail.view.xml [hunk 1]
```diff
@@ -70,7 +70,7 @@
                     <Button id="Delete" text="Delete" icon="sap-icon://delete" press="onRemovePress" />
                     <Button id="AddRule" text="Add Conformance Rule" icon="sap-icon://add-equipment" press="onAddConformanceRulePress" />
                     <Button id="Edit" text="Edit" icon="sap-icon://edit" press="onEditPress" enabled="{/editingEnabled}"
-                            tooltip="{= ${/editingEnabled} ? 'Edit current dataset' : 'Editing old version is disabled'}" />
+                            tooltip="{= ${/editingEnabled} ? 'Edit current dataset' : 'Editing old or locked version is disabled'}" />
```
Comment (benedeki) [inline menas/ui/components/dataset/datasetDetail.view.xml:73]: I would change the text
```suggestion
                            tooltip="{= ${/editingEnabled} ? 'Edit current dataset' : 'Editing old or locked version is not allowed'}" />
```

File: menas/ui/components/mappingTable/mappingTableDetail.view.xml [hunk 2]
```diff
@@ -96,7 +96,7 @@
                             press="onAddDefaultPress"/>
                     <Button id="Edit" text="Edit" icon="sap-icon://edit" press="onEditPress"
                             enabled="{/editingEnabled}"
-                            tooltip="{= ${/editingEnabled} ? 'Edit current mapping table' : 'Editing old version is disabled'}" />
+                            tooltip="{= ${/editingEnabled} ? 'Edit current mapping table' : 'Editing old or locked version is disabled'}" />
```
Comment (benedeki) [inline menas/ui/components/mappingTable/mappingTableDetail.view.xml:99]: DITTO with the wording:
```suggestion
                            tooltip="{= ${/editingEnabled} ? 'Edit current dataset' : 'Editing old or locked version is not allowed'}" />
```

File: menas/ui/components/schema/schemaDetail.view.xml [hunk 3]
```diff
@@ -147,7 +147,7 @@
                     <Button id="ExportStruct" text="Export Spark Struct" icon="sap-icon://action" press=".onExportStructPress"/>
                     <Button id="ExportFile" text="Export File" icon="sap-icon://action" press=".onExportFilePress"/>
                     <Button id="Edit" text="Edit" icon="sap-icon://edit" enabled="{/editingEnabled}"
-                            tooltip="{= ${/editingEnabled} ? 'Edit current schema' : 'Editing old version is disabled'}" />
+                            tooltip="{= ${/editingEnabled} ? 'Edit current schema' : 'Editing old or locked version is disabled'}" />
```
Comment (benedeki) [inline menas/ui/components/schema/schemaDetail.view.xml:150]: And third time the charm (also feel free to further improve on the text):
```suggestion
                            tooltip="{= ${/editingEnabled} ? 'Edit current dataset' : 'Editing old or locked version is not allowed'}" />
```

File: menas/ui/service/EntityService.js [hunk 4]
```diff
@@ -224,6 +224,10 @@ class EntityService {
     })
   }
 
+  canBeEdited(currentEntity, version) {
```
Comment (benedeki) [inline menas/ui/service/EntityService.js:227]: Nice 👍

**Review by benedeki — COMMENTED** [review]
Would it be too much to ask to add a key/;lock icon next to it's name when displaying a locked entity? 👼

**Review by dk1844 — COMMENTED** [review]
It looks good (I just read the code), but since #1950 also includes `dateLocked` and `userLocked`, shouldn't these values be shown, too?

**Review by dk1844 — DISMISSED** [review]
LGTM (just read the code)

**Review by dk1844 — APPROVED** [review]
Still LGTM (just read the code)

**Review by Zejnilovic — APPROVED** [review]
Just ran Menas and reviewed. Looks good.

Next ticket idea - Have the locked icon in the navigation as well

**Pull request merged**